### PR TITLE
WIP: Add comment parsing from CastXML

### PR DESCRIPTION
**Pull request by josephsnyder**
Todo: 

- [ ]  Determine if "endDocument" is the best place to link the comments.  Should it happen in visit_comment?
- [ ]  Determine best way to test CastXML for version before running test.  Should only execute on version >3.4
- [ ]  Determine if adding CastXML as PIP installation dependency during setup is desirable.

### CODE REVIEW

File: pygccxml/parser/scanner.py [hunk 1]
```diff
@@ -186,6 +193,19 @@ def xml_generator_from_xml_file(self):
     def read(self):
         xml.sax.parse(self.xml_file, self)
 
+    def _handle_comment(self, declaration):
+        comm_decl = self.__declarations.get(declaration.comment)
+        if comm_decl:
+            with open(self.__files.get(comm_decl.location.file_name, "w")) as file:
```
Comment (josephsnyder) [inline pygccxml/parser/scanner.py]: Should be "r" not "w".

**Comment by EricCousineau-TRI** [comment]
Rather than PR'ing to your own branch, could you instead PR to CastXML itself so we can see CI?
I'd rather do that (and have a potentially more messy discussion there) than discuss here then try to transport, etc.

**Comment by EricCousineau-TRI** [comment]
If you're not already aware, you can always use a draft PR in GitHub:
> image

(I believe CI still runs, which is great)

If it gets too noisy, then the PR itself could be closed and a new one re-opened.

**Comment by josephsnyder** [comment]
Closing this request in favor of a draft PR on pygccxml proper: https://github.com/CastXML/pygccxml/pull/130

**Pull request closed without merging**